mcs/class/System.Core/System.Security.Cryptography.X509Certificates/RSACertificateExtensions.cs

```diff
@@ -47,5 +47,10 @@ public static RSA GetRSAPublicKey(this X509Certificate2 certificate)
 				throw new ArgumentNullException("certificate");
 			return certificate.PublicKey.Key as RSA;
 		}
+		
+		public static X509Certificate2 CopyWithPrivateKey(this X509Certificate2 certificate, RSA privateKey) 
+		{ 
```

Comment:
This is not too difficult, I'll take care of that.

Reply:
@EgorBo I'll take care of this in https://github.com/mono/mono/pull/13037.

Reply:
Unfortunately, this is a lot more difficult than I expected, so I think we should keep the exception there for the moment.